components/script/dom/domrectlist.rs

```diff
@@ -57,3 +66,18 @@ impl DOMRectListMethods for DOMRectList {
         self.Item(index)
     }
 }
+
+impl<'a> Iterator for Iter<'a> {
```

Comment:
You can't return that from a function. However, it would probably be better to store `self.rects.iter()` in a field of `Iter` and then return `self.inner.next().map(|rect| Root::from_ref(&*rect))` from `next()`.